src/androidTest/java/com/couchbase/lite/AttachmentsTest.java

```diff
@@ -829,4 +834,49 @@ public int read() throws IOException {
             assertEquals(Status.STATUS_ATTACHMENT_ERROR, expected.getCBLStatus().getCode());
         }
     }
+
+    public void testGetAttachmentFromUnsavedRevision() throws Exception {
+        String attachmentName = "index.html";
+        String content  = "This is a test attachment!";
+
+        Document doc = createDocWithAttachment(database, attachmentName, content);
+        UnsavedRevision rev = doc.createRevision();
+        Attachment attachment = rev.getAttachment(attachmentName);
+        assertNotNull(attachment);
+
+        InputStream in = attachment.getContent();
+        assertNotNull(in);
+        assertEquals(IOUtils.toString(in, "UTF-8"), content);
+    }
+
+    public void testGzippedAttachments() throws Exception {
+        String attachmentName = "index.html";
+        byte content[]  = "This is a test attachment!".getBytes("UTF-8");
+
+        ByteArrayOutputStream byteOut = new ByteArrayOutputStream();
+        GZIPOutputStream gzipOut = new GZIPOutputStream(byteOut);
+        gzipOut.write(content);
+        gzipOut.close();
+        byte contentGzipped[] = byteOut.toByteArray();
+
+        Document doc = database.createDocument();
+        UnsavedRevision rev = doc.createRevision();
+        rev.setAttachment(attachmentName, "text/html", new ByteArrayInputStream(contentGzipped));
+        rev.save();
+
+        SavedRevision savedRev = doc.getCurrentRevision();
+        Attachment attachment = savedRev.getAttachment(attachmentName);
+
+        // As far as revision users are concerned their data is not gzipped
+        assertFalse(attachment.getGZipped());
+        MoreAsserts.assertEquals(content, IOUtils.toByteArray(attachment.getContent()));
+        assertFalse(attachment.getGZipped());
+
+        // But the it may be gzipped encoded internally
+        long sequence = savedRev.getSequence();
+        attachment = database.getAttachmentForSequence(sequence, attachmentName);
+        assertTrue(attachment.getGZipped());
+        MoreAsserts.assertEquals(contentGzipped, IOUtils.toByteArray(attachment.getContent()));
```

Comment:
Couchbase Lite unit tests are common code for Android and J2SE (Java). `MoreAsserts` is Android unit test. Please replace with common solution for both Android and J2SE.